tools/doc/markdown.js

```diff
@@ -2,18 +2,29 @@
 
 const visit = require('unist-util-visit');
 
+const referenceToLocalMdFile = /^(?![+a-z]+:)([^#?]+)\.md(#.+)?$/i;
+
 module.exports = {
-  replaceLinks
+  replaceLinks,
+  referenceToLocalMdFile,
 };
 
 function replaceLinks({ filename, linksMapper }) {
   return (tree) => {
     const fileHtmlUrls = linksMapper[filename];
 
+    visit(tree, (node) => {
+      if (node.url) {
+        node.url = node.url.replace(
+          referenceToLocalMdFile,
+          (_, filename, hash) => `${filename}.html${hash || ''}`
+        );
+      }
+    });
     visit(tree, 'definition', (node) => {
       const htmlUrl = fileHtmlUrls && fileHtmlUrls[node.identifier];
 
-      if (htmlUrl && typeof htmlUrl === 'string') {
+      if (typeof htmlUrl === 'string') {
```

Comment:
```
(htmlUrl && typeof htmlUrl === 'string')
```

You don't want to make sure that the variable exists before checking its type? I guess that's fine.

Reply:
The only difference is if/when `htmlUrl` is an empty string (which is falsy). I can remove this change as it is unrelated to the current PR.